app/scripts/metamask-controller.js

```diff
@@ -4454,6 +4456,38 @@ export default class MetamaskController extends EventEmitter {
 
       await this._addAccountsWithBalance(id);
```

Comment:
Should we move the new logic inside `_addAccountsWithBalance` instead? Cause right now, we won't be discovering Solana accounts after the onboarding is being done. (but that could be intended actually?)